clippy_lints/src/literal_representation.rs

```diff
@@ -365,54 +365,50 @@ impl LiteralDigitGrouping {
             if char::is_digit(firstch, 10);
             then {
 
+                let digit_info = match lit.kind {
+                        LitKind::Int(..) => DigitInfo::new(&src, false),
+                        LitKind::Float(..) => DigitInfo::new(&src, true),
+                        _ => return,
+                };
 
-        match lit.kind {
-            LitKind::Int(..) => {
-                // Lint integral literals.
-                let digit_info = DigitInfo::new(&src, false);
-                let _ = Self::do_lint(digit_info.digits, digit_info.suffix, in_macro).map_err(|warning_type| {
+                let result = (|| {
+                    match lit.kind {
+                        LitKind::Int(..) => {
+                            Self::do_lint(digit_info.digits, digit_info.suffix, in_macro)?;
+                        },
+                        LitKind::Float(..) => {
+                            // Separate digits into integral and fractional parts.
+                            let parts: Vec<&str> = digit_info
+                                .digits
+                                .split_terminator('.')
+                                .collect();
+
+                            // Lint integral and fractional parts separately, and then check consistency of digit
+                            // groups if both pass.
+                            let integral_group_size = Self::do_lint(parts[0], digit_info.suffix, in_macro)?;
+                            if parts.len() > 1 {
+                                // Lint the fractional part of literal just like integral part, but reversed.
+                                let fractional_part = &parts[1].chars().rev().collect::<String>();
+                                let fractional_group_size = Self::do_lint(fractional_part, None, in_macro)?;
+                                let consistent = Self::parts_consistent(integral_group_size,
+                                                                        fractional_group_size,
+                                                                        parts[0].len(),
+                                                                        parts[1].len());
+                                if !consistent {
+                                    return Err(WarningType::InconsistentDigitGrouping);
+                                };
+                            };
+                        },
+                        _ => (),
+                    }
+
+                    Ok(())
+                })();
```

Comment:
It's a DIY try block. I added it so that the errors can be gathered up into a result and handled afterwards. I tried to use the nightly `try_block` syntax but it doesn't allow `return Err(..)`.